### ISSUE TITLE
what is the argmax value? 

### ISSUE THREAD

**Issue by LinJM** [wwjwhen/Watch-Attend-and-Spell-tensorflow-version#2]
https://github.com/wwjwhen/Watch-Attend-and-Parse-tensorflow-version/blob/165469c6d04e758de1db8283224accc94490bc69/model-single-GPU.py#L476

Hello, @wwjwhen ，I found a uninitialized variable “argmax”， I now set to True? Does it OK?

**Comment by wwjwhen** [comment]
The decoding method is beam search instead of  simple greedy algorithm, if use stochastic and argmax, the   decoding just chooses the highest probability at each single time step, if we use beam search, the decoding considers the whole process from t0 to tn-1, which is more reasonable. BTW, I have not tested with greedy approach, so I can not give you specific conclusion.

**Comment by ducanh841988** [comment]
@LinJM  Do you successfully train model?
I can run the training code, but it is not coverage.
WER in the validation set does not reduce.
I use python 3.7 and tensorflow 1.6
@wwjwhen do you have any suggestion?
